gcloud/bigtable/test_client.py

```diff
@@ -523,3 +579,38 @@ def __enter__(self):
     def __exit__(self, exc_type, exc_val, exc_tb):
         self._exited.append((exc_type, exc_val, exc_tb))
         return True
+
+    def __getattr__(self, name):
+        # We need not worry about attributes set in constructor
+        # since __getattribute__ will handle them.
+        return _MethodMock(name, self)
+
+
+class _MethodMock(object):
+    """Mock for :class:`grpc.framework.alpha._reexport._UnaryUnarySyncAsync`.
+
+    May need to be callable and needs to (in our use) have an
+    ``async`` method.
+    """
+
+    def __init__(self, name, factory):
+        self._name = name
+        self._factory = factory
+
+    def async(self, *args, **kwargs):
+        """Async method meant to mock a gRPC stub request."""
+        self._factory.method_calls.append((self._name, args, kwargs))
+        curr_result = self._factory.results[0]
+        self._factory.results = self._factory.results[1:]
```

Comment:
I kind of prefer the "atomic" version using unpacking, which keeps from assigning anything at all in the case of an exception.

```python
    curr_result, self._factory-results = self._factory.results[0], self._factory.results[1:]
```